server/controllers/MiscController.js

```diff
@@ -52,43 +50,29 @@ class MiscController {
       return res.status(500).send(`Invalid post data`)
     }
 
-    // For setting permissions recursively
-    let outputDirectory = ''
-    let firstDirPath = ''
-
-    if (library.isPodcast) { // Podcasts only in 1 folder
-      outputDirectory = Path.join(folder.fullPath, title)
-      firstDirPath = outputDirectory
-    } else {
-      firstDirPath = Path.join(folder.fullPath, author)
-      if (series && author) {
-        outputDirectory = Path.join(folder.fullPath, author, series, title)
-      } else if (author) {
-        outputDirectory = Path.join(folder.fullPath, author, title)
-      } else {
-        outputDirectory = Path.join(folder.fullPath, title)
-      }
-    }
-
-    if (await fs.pathExists(outputDirectory)) {
-      Logger.error(`[Server] Upload directory "${outputDirectory}" already exists`)
-      return res.status(500).send(`Directory "${outputDirectory}" already exists`)
-    }
+    // Podcasts should only be one folder deep
+    const outputDirectoryParts = library.isPodcast ? [title] : [author, series, title]
+    // `.filter(Boolean)` to strip out all the potentially missing details (eg: `author`)
+    // before sanitizing all the directory parts to remove illegal chars and finally prepending
+    // the base folder path
+    const cleanedOutputDirectoryParts = outputDirectoryParts.filter(Boolean).map(part => sanitizeFilename(part))
+    const outputDirectory = Path.join(...[folder.fullPath, ...cleanedOutputDirectoryParts])
```

Comment:
Did a bit of a refactor here to slim things down while keeping things readable (imo). This should be functionally identical to the old code, save for my sanitization changes

Reply:
Ah I didn't know about `filter(Boolean)`, that is a nice shortcut!